src/hooks/use-favorite-movie-toggle.js

```diff
@@ -0,0 +1,35 @@
+import {useState, useCallback, useEffect} from 'react';
+
+import OperationStatus from '../constants/operation-status';
+import provider from '../providers/provider';
+
+const useFavoriteMovieToggle = (movieId) => {
+  const [status, setStatus] = useState(OperationStatus.UNSET);
+  const [isActive, setActive] = useState();
+  const [cancellation] = useState(provider.getCancellation());
```

Comment:
useRef тут семантически больше подходит
```suggestion
  const cancelation = useRef(provider.getCancellation()).current;
```